DiscordChatExporter.Core/Discord/DiscordClient.cs

```diff
@@ -200,6 +200,58 @@ private async ValueTask<TokenKind> GetTokenKindAsync(CancellationToken cancellat
         return Guild.Parse(response);
     }
 
+    public async IAsyncEnumerable<Data.Thread> GetGuildChannelThreadsAsync(
+        string channelId,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        int currentOffset = 0;
+
+        var completeThreads = new List<JsonElement>();
+
+        while (true)
+        {
+            var url = new UrlBuilder()
+                .SetPath($"channels/{channelId}/threads/search")
+                .SetQueryParameter("offset", currentOffset.ToString())
+                .Build();
+            try
+            {
+                var response = await GetJsonResponseAsync(url, cancellationToken);
+
+                completeThreads.AddRange(response.GetProperty("threads").EnumerateArray());
```

Comment:
Here's how you should do it instead:

```suggestion
                foreach (var threadJson in response.GetProperty("threads").EnumerateArray())
                		yield return Thread.Parse(threadJson);
```

Reply:
Resolved in 80cfb15.